lib/fastlane/plugin/appcenter/actions/appcenter_fetch_version_number.rb

```diff
@@ -1,45 +1,58 @@
-require 'json'
-require 'net/http'
-require 'fastlane_core/ui/ui'
+require "json"
+require "net/http"
+require "fastlane_core/ui/ui"
 
 module Fastlane
   UI = FastlaneCore::UI unless Fastlane.const_defined?("UI")
 
   module Actions
     class AppcenterFetchVersionNumberAction < Action
       def self.description
-        "Fetches the latest version number of an app from App Center"
+        "Fetches the latest version number of an app or the last build number of a version from App Center"
       end
 
       def self.authors
-        ["jspargo", "ShopKeep"]
+        ["jspargo", "ShopKeep", "Qutaibah"]
       end
 
       def self.run(params)
         api_token = params[:api_token]
         app_name = params[:app_name]
         owner_name = params[:owner_name]
+        version = params[:version]
 
         releases = Helper::AppcenterHelper.fetch_releases(
           api_token: api_token,
           owner_name: owner_name,
-          app_name: app_name
+          app_name: app_name,
         )
 
         UI.abort_with_message!("No versions found for '#{app_name}' owned by #{owner_name}") unless releases
-        sorted_release = releases.sort_by { |release| release['id'] }.reverse!
-        latest_release = sorted_release.first
+
+        sorted_releases = releases
+
+        if version.nil?
+          sorted_releases = releases.sort_by { |release| release["id"] }
+        else
+          sorted_releases = releases.select { |release| release["short_version"] == version }.sort_by { |release| release["id"] }
+        end
+
+        latest_release = sorted_releases.last
 
         if latest_release.nil?
-          UI.user_error!("This app has no releases yet")
+          if version.nil?
+            UI.user_error!("This app has no releases yet")
+            return nil
+          end
+          UI.user_error!("The provided version has no releases yet")
```

Comment:
Maybe also print out the requested version.

Reply:
Done